beater/config/java_attacher.go

```diff
@@ -0,0 +1,41 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package config
+
+// JavaAttacherConfig holds configuration information for running a java
+// attacher jarfile.
+type JavaAttacherConfig struct {
```

Comment:
I've almost overlooked it..

The config is sensitive to the ordering of the includes and excludes. See also https://www.elastic.co/guide/en/apm/agent/java/current/setup-attach-cli.html#setup-attach-cli-usage.

For example `--exclude-main ".*" --include-main opbeans` will exclude all JVMs, while `--include-main opbeans --exclude-main ".*"` includes only JVMs containing `opbeans` in their jar name.

Reply:
Given that it's order dependent, should we just accept an `args` string in the config instead of breaking it down into individual include/exclude flags? I'm not sure otherwise how else we would know how to order the flags for the attacher.

Reply:
That sounds good to me.

Reply:
It took me a while to find it again, but there's something similar in Fleet: https://github.com/elastic/beats/pull/23848

Applying the same principle here, the config could look like this:

```yaml
discovery-rules:
- rule: include # fleet uses allow/deny
  main: opbeans
- rule: include
  pid: 42
- rule: exclude
  main: ".*"
```

This would result in these arguments: `--include-main opbeans --include-pid 42 --exclude-main ".*"`

An alternative:

```yaml
discovery-rules:
- include-main: opbeans
- include-pid: 42
- exclude-main: ".*"
```

Reply:
I'd rather we have some structure to the options like this than accept arbitrary args. That would exclude any shenanigans around running the attacher in unexpected/unsupported ways by manually crafting the policy/config; the server should be strict about the ways in which it will run the attacher program.

Reply:
@axw / @simitt do you have a preference between @felixbarny's first and second suggestions? I'm leaning towards the second option, but am not opposed to option 1

Reply:
No strong opinion, either works for me and they will be hidden behind a layer of Kibana anyway - so whichever is simpler (probably 2?)